src/status_im/hardwallet/core.cljs

```diff
@@ -0,0 +1,57 @@
+(ns status-im.hardwallet.core
+  (:require [status-im.utils.platform :as platform]
+            [status-im.utils.config :as config]
+            [status-im.react-native.js-dependencies :as js-dependencies]
+            [re-frame.core :as re-frame]))
+
+(defn check-nfc-support []
+  (when config/hardwallet-enabled?
+    (.. js-dependencies/nfc-manager
+        -default
+        isSupported
+        (then #(re-frame/dispatch [:hardwallet.callback/check-nfc-support-success %])))))
+
+(defn check-nfc-enabled []
+  (when platform/android?
+    (.. js-dependencies/nfc-manager
+        -default
+        isEnabled
+        (then #(re-frame/dispatch [:hardwallet.callback/check-nfc-enabled-success %])))))
+
+(defn set-nfc-support [supported? {:keys [db]}]
+  {:db (assoc-in db [:hardwallet :nfc-supported?] supported?)})
+
+(defn set-nfc-enabled [enabled? {:keys [db]}]
+  {:db (assoc-in db [:hardwallet :nfc-enabled?] enabled?)})
+
+(defn open-nfc-settings []
+  (when platform/android?
+    (.. js-dependencies/nfc-manager
+        -default
+        goToNfcSetting)))
+
+(defn navigate-to-connect-screen []
+  {:dispatch                     [:navigate-to :hardwallet/connect]
```

Comment:
to check with @rasom as well but I think this is also something that can be done with `(navigation/navigate-to ...)`